integration/ledger/snapshot_test.go

```diff
@@ -0,0 +1,1049 @@
+		setup = initAndStartMultiChannelNetwork()
+
+		helper = &marblesTestHelper{
+			networkHelper: &networkHelper{
+				Network:   setup.network,
+				orderer:   setup.orderer,
+				peers:     setup.peers,
+				testDir:   setup.testDir,
+				channelID: setup.channelID,
+			},
+		}
+
+		legacyChaincode = nwo.Chaincode{
+			Name:        "marbles",
+			Version:     "0.0",
+			Path:        chaincodePathWithIndex,
+			Ctor:        `{"Args":[]}`,
+			Policy:      `OR ('Org1MSP.member','Org2MSP.member','Org3MSP.member','Org4MSP.member')`,
+			PackageFile: filepath.Join(setup.testDir, "marbles_legacy.tar.gz"),
+		}
+
+		newlifecycleChaincode = nwo.Chaincode{
+			Name:            "marbles",
+			Version:         "0.0",
+			Path:            components.Build(chaincodePathWithIndex),
+			Lang:            "binary",
+			CodeFiles:       filesWithIndex,
+			PackageFile:     filepath.Join(setup.testDir, "marbles.tar.gz"),
+			SignaturePolicy: `OR ('Org1MSP.member','Org2MSP.member','Org3MSP.member','Org4MSP.member')`,
+			Sequence:        "1",
+			Label:           "marbles",
+		}
+	})
+
+	AfterEach(func() {
+		setup.cleanup()
+		os.RemoveAll(setup.network.RootDir)
+	})
+
+	// Below test does the following when using leveldb as the statedb:
+	// - deploy legacy chaincode
+	// - create snapshots on 2 peers and verify they are same
+	// - bootstrap peers (in existing org and new org) from the snapshot
+	// - verify functions (chaincode invocation/query, channel config update, join new channel, qscc, history)
+	// - upgrade to new lifecycle chaincode
+	// - create a new snapshot from a peer bootstrapped from a snapshot
+	// - bootstrap peers (in existing org and new org) from the new snapshot
+	// - verify functions (chaincode invocation/query, channel config update, join new channel, qscc, history)
+	// - install a new chaincode and verify invocation/query
+	When("levelDB is used as stateDB", func() {
+		It("generates snapshots and bootstraps from snapshot with stateleveldb", func() {
+			org1peer0 := setup.network.Peer("Org1", "peer0")
+			org2peer0 := setup.network.Peer("Org2", "peer0")
+
+			By("deploying legacy chaincode on peers with stateleveldb")
+			nwo.DeployChaincodeLegacy(setup.network, testchannelID, setup.orderer, legacyChaincode)
+
+			height := helper.getLedgerHeight(org1peer0)
+			By(fmt.Sprintf("adding marbles at height %d", height))
+			testKey := "marble-0"
+			channelPeers := setup.network.PeersWithChannel(testchannelID)
+			helper.invokeMarblesChaincode(legacyChaincode.Name, org1peer0, "initMarble", "marble-0", "blue", "35", "tom")
+			nwo.WaitUntilEqualLedgerHeight(setup.network, testchannelID, height+1, channelPeers...)
+			helper.invokeMarblesChaincode(legacyChaincode.Name, org1peer0, "initMarble", "marble-1", "blue", "35", "tom")
+			nwo.WaitUntilEqualLedgerHeight(setup.network, testchannelID, height+2, channelPeers...)
+
+			By("getting txid from a block before snapshot is generated")
+			txid := getTxIDFromLastBlock(setup.network, org1peer0)
```

Comment:
This appears a round-about way to get the TxID, it will be preferred to create a signed proposal explicitly - (I believe that there is a function for this). Perhaps, you can modify the function "invokeMarblesChaincode" to returns the TxID or have another variation of this funciton.